frontend/src/hooks/useModal.tsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useRef } from 'react';
+
```

Comment:
Core file, this is the modal mechanism. It tracks and handle events such as escaping. And then then also scroll-locking ;).

Reply:
and i forgot focus.